main/src/cgeo/geocaching/EditWaypointActivity.java

```diff
@@ -333,29 +333,29 @@ public void onNothingSelected(final AdapterView<?> parent) {
             }
         });
 
-        binding.type.setSelection(getDefaultWaypointType());
+        binding.type.setSelection(getDefaultWaypointTypePosition(wpAdapter));
         binding.type.setVisibility(View.VISIBLE);
     }
 
-    private int getDefaultWaypointType() {
+    private int getDefaultWaypointTypePosition(final ArrayAdapter<WaypointType> wpAdapter) {
         // potentially restore saved instance state
         if (waypointTypeSelectorPosition >= 0) {
             return waypointTypeSelectorPosition;
         }
 
         // when editing, use the existing type
         if (waypoint != null) {
-            return POSSIBLE_WAYPOINT_TYPES.indexOf(waypoint.getWaypointType());
+            return wpAdapter.getPosition(waypoint.getWaypointType());
```

Comment:
What should happen if (for whatever reason) the waypoint type is not found in the adapter? Returning -1 ok?

Reply:
Good question - will show what the old return-value was and was the new one is.

Reply:
ok, then it is finer